### PR TITLE
refactor logging to make idiomatic use of ruby log levels

### PR DESCRIPTION
**Pull request by snatchev**
This PR proposes a change to the logging levels to be more idiomatic and have less conditional checks on environment variables.

The default log-level will be set at `WARN` and any calls to `logger.warn` will be printed.
If `DEBUG` or `FASTLANE_CI_VERBOSE` are set, then the log-level is lowered to the most verbose setting `DEBUG`. The difference being that `FASTLANE_CI_VERBOSE` will have extra thread IDs.

Logging about normal system operation should be done with `logger.info`

This PR also renames FASTLANE_CI_SUPER_VERBOSE -> FASTLANE_CI_VERBOSE. IMO, there should be a verbose to start with before adding a super version.

### CODE REVIEW

File: shared/logging_module.rb [hunk 1]
```diff
@@ -14,16 +37,23 @@ def logger
             class_name = self.name.split("::").last
           end
 
+          progname ||= class_name
```
Comment (snatchev) [inline shared/logging_module.rb]: When using the logger's block syntax, and providing a `progname` we should preserve the expected behavior of having that appear in the log line.

File: shared/logging_module.rb [hunk 2]
```diff
@@ -14,16 +37,23 @@ def logger
...
             thread_id = " #{Thread.current[:thread_id]}" unless Thread.current[:thread_id].nil?
           end
 
-          "[#{datetime.strftime('%Y-%m-%d %H:%M:%S')}#{thread_id}] #{severity} #{class_name}:  #{msg}\n"
+          "[#{datetime.strftime('%Y-%m-%d %H:%M:%S')}#{thread_id}] #{severity} #{progname}:  #{msg}\n"
         end
+
+        logger.level = :warn
+        logger.level = :debug if ENV['DEBUG']
+        logger.level = :debug if ENV['FASTLANE_CI_SUPER_VERBOSE']
```
[the remark this answers is not in the document]
Comment (snatchev) [reply shared/logging_module.rb]: Currently `FASTLANE_CI_SUPER_VERBOSE` just configs whether or not the Thread id's are included in the log lines. What if we make the inclusion of Thread id dependent on `DEBUG=1` and add a different env `LOG_LEVEL` that sets the desired log level. I think this kind of decouples the concerns a little better. Just a thought, I am open to keeping as-is too.

[the remark this answers is not in the document]
Comment (KrauseFx) [reply shared/logging_module.rb]: There were other things also, like https://github.com/fastlane/ci/pull/148/files#diff-abfd6045c8ab458dd65dec0a3ad9c66aL113

[the remark this answers is not in the document]
Comment (taquitos) [reply shared/logging_module.rb]: How about `FASTLANE_CI_VERBOSE`? That's really what we're talking about 😄

File: shared/logging_module.rb [hunk 3]
```diff
@@ -14,16 +37,23 @@ def logger
             class_name = self.name.split("::").last
           end
 
+          progname ||= class_name
+
           thread_id = ""
 
-          if ENV["FASTLANE_CI_SUPER_VERBOSE"]
+          if ENV['FASTLANE_CI_SUPER_VERBOSE']
```
Comment (taquitos) [inline shared/logging_module.rb]: We use `"` for all strings to keep it simple

File: shared/logging_module.rb (hunk 1 above)
Comment (taquitos) [reply shared/logging_module.rb]: 🏆

File: shared/logging_module.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (taquitos) [reply shared/logging_module.rb]: Decoupling is the name of the game 🚀

**Comment by taquitos** [comment]
@snatchev this is great, thank you!

File: shared/logging_module.rb (hunk 3 above)
Comment (snatchev) [reply shared/logging_module.rb]: I neglected to run rubocop. :womp:

**Comment by snatchev** [comment]
did the rename as suggested by https://github.com/fastlane/ci/pull/148#discussion_r169216426

**Comment by taquitos** [comment]
🎉

**Pull request merged**